app/controllers/results_controller.rb

```diff
@@ -134,6 +133,19 @@ def edit
     end
   end
 
+  def run_tests
+    grouping_id = params[:grouping_id]
+    begin
+      AutomatedTestsHelper.request_a_test_run(grouping_id,
+                                              'request',
+                                              @current_user)
+    rescue => e
+      # TODO: really shouldn't be leaking error if student.
+      return e.message
```

Comment:
This isn't the appropriate behaviour for a method which is directly responding to a request. Instead, do `flash_message(:error, e.message)`